gh-notify v0.1: a magit/forge veneer for github notifications

anticomputer opened the pull request:
### Brief summary of what the package does

gh-notify is a thin ui veneer on top of magit/forge porcelain for juggling large amounts of GitHub notifications at speed.

### Direct link to the package repository

https://github.com/anticomputer/gh-notify

### Your association with the package

maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm with `x`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks!  Looks good -- also looks like this is already a popular package, so congrats.

You may have noticed byte-compile warnings along these lines:
```
gh-notify.el:303:30:Warning: `forge-notification' is an obsolete variable (as
    of 25.1); use 'forge-notification instead
```
which is non-blocking but I'll list it fyi.

And a couple of other issues, from most important to least:
- Add a LICENSE file that GitHub can detect (e.g. no markup) if possible
- It looks like the default recipe will be equivalent in this case: `(gh-notify :repo "anticomputer/gh-notify" :fetcher github)`
- This is a non-blocker since you seem to have caught everything, but if you use `define-derived-mode` instead of `defun gh-notify-mode` a lot of the boilerplate will disappear; you're probably already aware of this though
- gh-notify.el#L722: It's safer to sharp-quote function names; use `#'` (minor, not a blocker)

anticomputer commented:
> Thanks! Looks good -- also looks like this is already a popular package, so congrats.
> 
> You may have noticed byte-compile warnings along these lines:
> 
> ```
> gh-notify.el:303:30:Warning: `forge-notification' is an obsolete variable (as
>     of 25.1); use 'forge-notification instead
> ```
> 
> which is non-blocking but I'll list it fyi.
> 
> And a couple of other issues, from most important to least:
> 
> * Add a LICENSE file that GitHub can detect (e.g. no markup) if possible
> * It looks like the default recipe will be equivalent in this case: `(gh-notify :repo "anticomputer/gh-notify" :fetcher github)`
> * This is a non-blocker since you seem to have caught everything, but if you use `define-derived-mode` instead of `defun gh-notify-mode` a lot of the boilerplate will disappear; you're probably already aware of this though
> * gh-notify.el#L722: It's safer to sharp-quote function names; use `#'` (minor, not a blocker)

@riscy thanks, appreciate the review! I've addressed your feedback where possible in the latest commits to gh-notify as well as on this PR:

- I was unable to reproduce the obsolete variable warnings on byte-compile, I did address some doc-string length ones and I get zero warnings on my end now, but if there's anything specific that still looks concerning please let me know
- I've added a dedicated LICENSE file with the main BSD 2-clause as well as the other required license reproductions
- I've updated the melpa recipe in the PR to the default form
- I've sharp-quoted the string function name
- re: define-derived-mode, point taken, will look to simplify in a future release

riscy commented:
Thanks, I'll merge.  The byte-compile might be due to me installing the very latest version of forge -- and it looks like GitHub is having trouble detecting your LICENSE file, so you should see if you can put it in a standard format (detectable by https://github.com/licensee/licensee) but that's not a blocker.
